network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalImpl.java

```diff
@@ -249,9 +244,6 @@ private boolean connectNodeBreaker(Predicate<Switch> isTypeSwitchToOperate) {
             return false;
         }
 
-        // The switches are now closed
-        closeSwitches(index, switchesToClose);
-
         return true;
```

Comment:
I would rather put that in the if branch... closeSwitches method does setOpen and update the resource so I think it's important to keep it

Reply:
in closeSwitches method
```
// Close
switchImpl.setOpen(false);
// Update the resource
index.updateSwitchResource(switchImpl.getResource());
```

In the disconnectNodeBreaker method 
```
// Open the switches
switchesToClose.forEach(sw -> sw.setOpen(false));
```

So we don't do the updateSwitchResource

Reply:
See the method setOpen in SwitchImpl : it does all the things ...